src/main/java/hudson/plugins/copyartifact/CopyArtifact.java

```diff
@@ -371,15 +451,25 @@ private boolean canReadFrom(Job<?, ?> job, AbstractBuild<?, ?> build) {
         return b;
     }
 
-    // retrieve the "folder" (jenkins root if no folder used) for this build
-    private ItemGroup getItemGroup(AbstractBuild<?, ?> build) {
-        ItemGroup group = build.getProject().getRootProject().getParent();
-        return group;
+    private Job<?, ?> getRootJob(Job<?, ?> job) {
+        if (job instanceof AbstractProject) {
+            return ((AbstractProject<?,?>)job).getRootProject();
+        } else {
+            return job;
+        }
+    }
 
+    // retrieve the "folder" (jenkins root if no folder used) for this build
+    private ItemGroup getItemGroup(Run<?, ?> build) {
+        if (build instanceof AbstractBuild) {
+            return ((AbstractBuild) build).getProject().getRootProject().getParent();
+        } else {
+            return build.getParent().getParent();
+        }
     }
 
 
-    private boolean perform(Run src, AbstractBuild<?,?> dst, String expandedFilter, String expandedExcludes, FilePath targetDir,
+    private boolean perform(Run src, Run<?,?> dst, String expandedFilter, String expandedExcludes, FilePath targetDir,
```

Comment:
I just figured that since in the public overload of `perform` we are changing the signature to use a meaningful `AbortException` rather than the deprecated `false`, it makes sense to change it here to match.